### PR TITLE
feature: get event type by topic

### PR DESCRIPTION
**Pull request by adyach**
Admin api to get ET by topic + fixed errors swagger file

### CODE REVIEW

File: api/nakadi-event-bus-api.yaml [hunk 1]
```diff
@@ -1617,6 +1618,34 @@ paths:
           schema:
             $ref: '#/definitions/Problem'
 
+  /topics/{name}:
+    get:
+      tags:
+        - topics-api
+      description: |
+        Retrieves the event type by provided topic.
+        It searches for event type using the latest active timeline.
```
Comment (antban) [inline api/nakadi-event-bus-api.yaml]: Why only latest

Comment (adyach) [inline api/nakadi-event-bus-api.yaml]: It was my requirement

File: api/nakadi-event-bus-api.yaml [hunk 2]
```diff
@@ -1617,6 +1618,34 @@ paths:
...
+        - name: name
+          in: path
+          description: Name of the topic
+          type: string
+          required: true
+      responses:
+        '200':
+          description: Event type wihch is represented by provided topic
+          schema:
+            $ref: '#/definitions/EventType'
```
Comment (antban) [inline api/nakadi-event-bus-api.yaml]: Theoretically it should be an array

File: src/main/java/org/zalando/nakadi/service/TopicsService.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        this.timelineDbRepository = timelineDbRepository;
+        this.eventTypeRepository = eventTypeRepository;
+    }
+
+    public EventType getEventTypeByTopic(final String topic, final Client client) {
+        if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            throw new ForbiddenAccessException("Request is forbidden for user " + client.getClientId());
+        }
+
+        final Optional<String> etName = timelineDbRepository.listTimelinesOrdered().stream()
```
Comment (antban) [inline src/main/java/org/zalando/nakadi/service/TopicsService.java]: I understand that this operation is really rare, but you will fetch all Timelines in db. At some point nakadi will just die.

Comment (adyach) [inline src/main/java/org/zalando/nakadi/service/TopicsService.java]: My point was that there are no timelines in database

File: api/nakadi-event-bus-api.yaml [hunk 4]
```diff
@@ -1617,6 +1618,34 @@ paths:
           schema:
             $ref: '#/definitions/Problem'
 
+  /topics/{name}:
+    get:
```
Comment (v-stepanov) [inline api/nakadi-event-bus-api.yaml]: This is not very RESTful. When I do GET /topics/myTopic I expect to get some topic object but instead I get EventType (for which there is a separate resource)
So the endpoint should be more like "get event-type by topic"

I think the endpoint
`GET /event-types?topic=9bc91968-101b-435b-b697-33d47f8c69c4` will fit more

File: src/main/java/org/zalando/nakadi/controller/TopicsController.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.zalando.nakadi.exceptions.runtime.NoTopicFoundException;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.service.TopicsService;
+import org.zalando.problem.Problem;
+import org.zalando.problem.spring.web.advice.Responses;
+
+import javax.ws.rs.core.Response;
+
+@RestController
+@RequestMapping(value = "/topics/")
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/controller/TopicsController.java]: I am missing security configuration for the new endpoint. (And the test for that)

**Comment by lmontrieux** [comment]
@adyach are you still working on this?

**Comment by adyach** [comment]
@lmontrieux yep, give me sometime for that.

**Pull request closed without merging**